### PR TITLE
style: add prettier

### PR DESCRIPTION
**Pull request by henryiii**
This adds prettier to our pre-commit, which styles yaml files (and potentially markdown & html if we have any eventually, even including yaml code blocks in markdown).

### CODE REVIEW

**Review by layday — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ on:
     branches:
       - main
     paths-ignore:
-      - 'docs/**'
-      - '*.md'
+      - "docs/**"
+      - "*.md"
```
Comment (gaborbernat) [inline .github/workflows/test.yml:8]: I think we need to remove this as prettier formats the markdown document, no?

Comment (henryiii) [reply .github/workflows/test.yml:8]: But pre-commit runs separately (on pre-commit.ci), it's not part of this job?

Comment (henryiii) [reply .github/workflows/test.yml:8]: (Same issue would be true with end-of-file checks, etc)

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**